models/utilities/default_model_mod.f90

```diff
@@ -283,6 +302,157 @@ subroutine pert_model_copies(state_ens_handle, ens_size, pert_amp, interf_provid
 
 end subroutine pert_model_copies
 
+!--------------------------------------------------------------------
+
+!> Parses the character table that was read in from
+!> model_nml:model_state_variables and returns a
+!> state_var_type state_vars with nvars ; netcdf variable names ;
+!> qtys (kinds) ; clamp values ; updates
+!> that there are valid entries for the dart_kind.
+!
+!> Verifies that the namelist entry was filled in correctly, and checks
+!> that there are valid entries for the dart_kind.
+
+function parse_variables_clamp(vars_table) result(state_vars)
+
+character(len=*), intent(in) :: vars_table(MAX_STATE_VARIABLE_FIELDS_CLAMP)
+type(state_var_type) :: state_vars
+
+character(len=NF90_MAX_NAME) :: netcdf_var_name, dart_qty_str, update
+character(len=256) :: string1, string2
+integer :: i, ivar
+
+! Loop through the variables array to get the actual count of the number of variables
+do ivar = 1, MAX_STATE_VARIABLES
+   ! If the first element in the row is an empty string, the loop has exceeded the extent of the variables
+   if (vars_table(5*ivar-4) == '') then
+      state_vars%nvars = ivar-1
+      exit
+   endif
+enddo
+
+! Allocate the arrays in the var derived type
+allocate(state_vars%netcdf_var_names(state_vars%nvars), state_vars%qtys(state_vars%nvars), state_vars%clamp_values(state_vars%nvars, 2), state_vars%updates(state_vars%nvars))
+
+RowsLoop : do i = 1, state_vars%nvars
+
+   netcdf_var_name = trim(vars_table(5*i-4))
+   state_vars%netcdf_var_names(i) = trim(netcdf_var_name)
+
+   dart_qty_str = trim(vars_table(5*i-3))
+   call to_upper(dart_qty_str)
+   ! Make sure DART qty is valid
+   state_vars%qtys(i) = get_index_for_quantity(dart_qty_str)
+   if( state_vars%qtys(i) < 0 ) then
+      write(string1,*) 'The quantity specified in the &model_nml "', dart_qty_str, '", is not present in obs_kind_mod.f90'
```

Comment:
This write to the string will segfault, because string1 is 256 & dart_qty_str is 256 
256 characters in dart_qty_str + 85 characters in the message > 256 in string1

```suggestion
      write(string1,*) 'The quantity specified in the &model_nml "', trim(dart_qty_str), '", is not present in obs_kind_mod.f90'
```